trie/trie.go

```diff
@@ -590,7 +568,7 @@ func (t *Trie) Commit(collectLeaf bool) (common.Hash, *NodeSet) {
 		t.root = hashedNode
 		return rootHash, nil
 	}
-	h := newCommitter(t.owner, t.tracer, collectLeaf)
+	h := newCommitter(t.owner, t.accessList, collectLeaf)
 	newRoot, nodes := h.Commit(t.root)
```

Comment:
This loop here is a bit wonky. The `t.accessList` is passed to the new `committer`, but the committer ignores it totally, and just passes it along to the `NodeSet`. 
The `h.Commit` operation returns said `NodeSet` in the form of `nodes`. And we deliver `nodes` to the caller. 

It looks a bit odd that we pass the accesslist down into `newComitter` only for that thing to bolt it onto the `nodes`. We could do that ourselves e.g 

```
h = newComitter(t, owner, collecLeaf)
newRoot, nodes := h.Commit(t.root)
nodes.setAccessList(t.accessList)
```
Alternatively, we could handle them separately: 

```
h = newComitter(t, owner, collecLeaf)
newRoot, nodes := h.Commit(t.root)
t.root = newRoot
return rootHash nodes, accessList
```

Reply:
The second option might not be my preference. `AccessList` or other information about trie might be changed in the future(we may need more/less), put all of them in nodeSet can avoid touching interface too frequently.

I am fine with the first option.